docs/design/2021-08-23-history-hot-regions.md

```diff
@@ -0,0 +1,269 @@
+      HotDegree      int       `json:"hot_degree"`
+      ByteRate       float64   `json:"flow_bytes"`
+      KeyRate        float64   `json:"flow_keys"`
+      QueryRate      float64   `json:"flow_query"`
+      AntiCount      int       `json:"anti_count"`
+      LastUpdateTime time.Time `json:"last_update_time"`
+    }
+    ```
+
+1. After TiDB catch the response, it  fetch the `START_KEY` and `END_KEY` of the hot region from region cache or PD by `REGION_ID` to decode the corresponding schema information like：`DB_NAME`, `TABLE_NAME`,  `TABLE_ID`,  `INDEX_NAME`, `INDEX_ID`.
+
+1. TiDB return the hotspot region row to the upper caller.
+
+In addition, hot regions can also be obtained directly through [pd-ctl](https://docs.pingcap.com/zh/tidb/stable/pd-control#health).  
+
+### Table Header Design
+1. New table header
+  ```SQL
+  > USE information_schema;
+  > DESC tidb_hot_regions_history;
+  +-------------+-------------+------+------+---------+-------+
+  | Field       | Type        | Null | Key  | Default | Extra |
+  +-------------+-------------+------+------+---------+-------+
+  | UPDATE_TIME | timestamp(6)| YES  |      | NULL    |       | // new
+  | DB_NAME     | varchar(64) | YES  |      | NULL    |       |
+  | TABLE_NAME  | varchar(64) | YES  |      | NULL    |       |
+  | TABLE_ID    | bigint(21)  | YES  |      | NULL    |       |
+  | INDEX_NAME  | varchar(64) | YES  |      | NULL    |       |
+  | INDEX_ID    | bigint(21)  | YES  |      | NULL    |       |
+  | REGION_ID   | bigint(21)  | YES  |      | NULL    |       | 
+  | STORE_ID    | bigint(21)  | YES  |      | NULL    |       | // new
+  | PEER_ID     | bigint(21)  | YES  |      | NULL    |       | // new
+  | IS_LEARNER  | tinyint(1)  | NO   |      | 0       |       | // new
+  | IS_LEADER   | tinyint(1)  | NO   |      | 0       |       | // new
+  | TYPE        | varchar(64) | YES  |      | NULL    |       |
+  | HOT_DEGREE  | bigint(21)  | YES  |      | NULL    |       | // rename max_hot_degree to HOT_DEGREE
+  | FLOW_BYTES  | double      | YES  |      | NULL    |       |
+  | KEY_RATE    | double      | YES  |      | NULL    |       | // new
+  | QUERY_RATE  | double      | YES  |      | NULL    |       | // new
+  +-------------+-------------+------+------+---------+-------+
+  | REGION_COUNT| bigint(21)  | YES  |      | NULL    |       | // deleted fields
+  ```
+  * Add `UPDATE_TIME` to support history.
+  * Add `STORE_ID` ,  `PEER_ID`,  `IS_LAEDER`to track the machine of region.
+  * Rename `MAX_HOT_DEGREE`  to `HOT_DEGREE` for precise meaning in history scenario.
+  * Add `IS_LEARNER` and `IS_LEADER` to show role of regions.
+  * Add `KEY_RATE` and `QUERY_RATE` for future expansion in hotspot determination dimensions.
+  * Remove `REGION_COUNT` for disuse and repeat with `STORE_ID`.
+
+2. Data size estimation 
+
+    one record size: 1 * 4B(timestamp) + 1*1B(tinyint) + 6 * 8B(bitint) + 3 * 8B(double)  + 4 * 64B(varchar(64)) = 333B
```

Comment:
why 4*64B?

Reply:
it's wrong, M of varchar(M) represents the maximum column length in characters (not bytes). The space occupied by a single character might differ for different character sets, form 1 to 4 bytes. it has ben addressed by new comment.
ref: [varchar type](https://docs.pingcap.com/tidb/stable/data-type-string#varchar-type)